engine/api/src/main/java/io/deephaven/engine/table/updateBySpec/EmaSpec.java

```diff
@@ -1,61 +1,46 @@
 package io.deephaven.engine.table.updateBySpec;
 
+import io.deephaven.annotations.SimpleStyle;
 import io.deephaven.engine.table.EmaControl;
 import io.deephaven.util.type.TypeUtils;
+import org.immutables.value.Value.Parameter;
+import org.immutables.value.Value.Immutable;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
  * A {@link UpdateBySpec} for performing an Exponential Moving Average across the specified columns
  */
-public class EmaSpec implements UpdateBySpec {
-    private final String timestampCol;
-    private final long timeScaleUnits;
-    private final EmaControl control;
-
+@Immutable
+@SimpleStyle
+public abstract class EmaSpec implements UpdateBySpec {
     public static EmaSpec ofTime(@NotNull final EmaControl control,
             @NotNull final String timestampCol,
             long timeScaleNanos) {
-        return new EmaSpec(control, timestampCol, timeScaleNanos);
+        return ImmutableEmaSpec.of(control, timestampCol, timeScaleNanos);
     }
 
     public static EmaSpec ofTicks(@NotNull EmaControl control, long tickWindow) {
-        return new EmaSpec(control, null, tickWindow);
+        return ImmutableEmaSpec.of(control, null, tickWindow);
     }
 
-    EmaSpec(@NotNull final EmaControl control, @Nullable final String timestampCol, final long timeScaleUnits) {
-        this.timestampCol = timestampCol;
-        this.control = control;
-        this.timeScaleUnits = timeScaleUnits;
-    }
+    @Parameter
+    public abstract EmaControl control();
 
-    public String getTimestampCol() {
-        return timestampCol;
-    }
-
-    public long getTimeScaleUnits() {
-        return timeScaleUnits;
-    }
+    @Parameter
+    @Nullable
+    public abstract String timestampCol();
 
-    public EmaControl getControl() {
-        return control;
-    }
-
-    @NotNull
-    @Override
-    public String describe() {
-        return "Ema(" + (timestampCol == null ? "TickBased" : timestampCol) + ", " + timeScaleUnits + ", " + control
-                + ")";
-    }
+    @Parameter
+    public abstract long timeScaleUnits();
 
     @Override
     public boolean applicableTo(@NotNull Class<?> inputType) {
```

Comment:
I find it useful to mark all methods implemented as final - not because I would ever expect Immutables to code-generate/overwrite them, but there may be some bytecode benefits. (I see you've taken that pattern that we've taken before w/ `walk`)